File: src/Benchmarking.java
import java.util.Random;
public class Benchmarking {
     private MetodosOrdenamiento metodosOrdenamiento;
    public Benchmarking(){
        //long inicioMillis=System.currentTimeMillis();
        //long inicioNano=System.nanoTime();
        //System.out.println(inicioMillis);
        //System.out.println(inicioNano);
        metodosOrdenamiento = new MetodosOrdenamiento();
        int[] arreglo = generarArregloAleatorio(1000000);
        Runnable tarea = () -> metodosOrdenamiento.burbujaTradicional(arreglo);

        double tiempoNano= medirConNanoTime(tarea);
        double tiempoMillis = medircurrentTimeMillis(tarea);
        System.out.println("Tiempo con nano time"+ tiempoNano + "segundos");
        System.out.println("Tiempo con currentTimeMillis:"+ tiempoMillis+"segundos");
    
    }


    private int[] generarArregloAleatorio( int tamano){
        int [] arreglo = new int [tamano];
        Random ramdom = new Random();
        for ( int i =0 ; i < tamano ; i++){
            arreglo[i] = ramdom.nextInt(100_000);
        }
        return arreglo;
    }

    public double medirConNanoTime(Runnable tarea){
        long inicio = System.nanoTime();
        tarea.run();
        long fin = System.nanoTime();
        return ( fin - inicio)/ 1_000_000_000.0;
    }

    public double medircurrentTimeMillis(Runnable tarea){
        long inicio = System.currentTimeMillis();
        tarea.run();
        long fin = System.currentTimeMillis();
        return(fin - inicio)/ 1000.0;
    }
}

File: bin/src_python/benchmarking.py
import random;
import time;
import metodos_ordenamiento as mo ;
class Benchmarking():
    def __init__(self):
        print('Bench inicializado')
        arreglo= self.build_arreglo(1000)
        self.mOrdenamiento= mo.MetodosOrdenamiento()
        tarea= lambda: self.mOrdenamiento.sortByBubble(arreglo)
        tiempoMillis = self.contar_con_current_time_milles(tarea)
        tiempoNano = self.contar_con_nano_time(tarea)
        print(f'Tiempo{tiempoMillis}')
        print(f'Tiempo{tiempoNano}')

    def build_arreglo(self,tamanio):
        array=[]
        for i in range(tamanio):
            numero = random.randint(0,99999)
            array.append(numero)
        return array   

    def contar_con_current_time_milles(self, tarea):
        x= time.time()
        tarea()
        fin = time.time()
        return fin-x


    def contar_con_nano_time(self,tarea):
        x= time.time_ns()
        tarea()
        fin = time.time_ns()
        return  fin -x

            